pkg/toolsets/toolsets.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package toolsets
+
+import "strings"
+
+// Toolset represents a group of related tools
+type Toolset string
+
+const (
+	// Core toolsets
+	ToolsetRegistry        Toolset = "registry"         // Public Terraform Registry
+	ToolsetRegistryPrivate Toolset = "registry-private" // Private registry (TFE/TFC)
+	ToolsetTerraform       Toolset = "terraform"        // TFE/TFC operations
+
+	// Special toolsets
+	ToolsetAll     Toolset = "all"
+	ToolsetDefault Toolset = "default"
+)
+
+type ToolsetMetadata struct {
+	ID          string
+	Description string
+}
+
+var (
+	MetadataAll = ToolsetMetadata{
+		ID:          string(ToolsetAll),
+		Description: "Special toolset that enables all available toolsets",
+	}
+	MetadataDefault = ToolsetMetadata{
+		ID:          string(ToolsetDefault),
+		Description: "Special toolset that enables the default toolset configuration",
+	}
+	MetadataRegistry = ToolsetMetadata{
+		ID:          string(ToolsetRegistry),
+		Description: "Public Terraform Registry (providers, modules, policies)",
+	}
+	MetadataRegistryPrivate = ToolsetMetadata{
+		ID:          string(ToolsetRegistryPrivate),
+		Description: "Private registry access (TFE/TFC private modules and providers)",
+	}
+	MetadataTerraform = ToolsetMetadata{
+		ID:          string(ToolsetTerraform),
+		Description: "HCP Terraform/TFE operations (workspaces, runs, variables, etc.)",
+	}
+)
+
+// AvailableToolsets returns all available toolsets
+func AvailableToolsets() []ToolsetMetadata {
+	return []ToolsetMetadata{
+		MetadataRegistry,
+		MetadataRegistryPrivate,
+		MetadataTerraform,
+	}
+}
+
+// DefaultToolsets returns the default set of enabled toolsets
+func DefaultToolsets() []string {
+	return []string{
+		string(ToolsetRegistry),
+	}
+}
+
+// GetValidToolsetIDs returns a map of all valid toolset IDs
+func GetValidToolsetIDs() map[string]bool {
+	validIDs := make(map[string]bool)
+	for _, ts := range AvailableToolsets() {
+		validIDs[ts.ID] = true
+	}
+	validIDs[MetadataAll.ID] = true
+	validIDs[MetadataDefault.ID] = true
+	return validIDs
+}
+
+// CleanToolsets sanitizes the user input
+func CleanToolsets(enabledToolsets []string) ([]string, []string) {
+	seen := make(map[string]bool)
+	result := make([]string, 0, len(enabledToolsets))
+	invalid := make([]string, 0)
+	validIDs := GetValidToolsetIDs()
+
+	for _, toolset := range enabledToolsets {
+		trimmed := strings.TrimSpace(toolset)
+		if trimmed == "" {
+			continue
+		}
+		if !seen[trimmed] {
+			seen[trimmed] = true
+			result = append(result, trimmed)
+			if !validIDs[trimmed] {
+				invalid = append(invalid, trimmed)
+			}
+		}
+	}
+
+	return result, invalid
+}
+
+func ExpandDefaultToolset(toolsets []string) []string {
```

Comment:
Similarly, here (and below in `ContainsToolset`) you have used `toolsets []string`, but we define a specific type `Toolset` for this above. To be consistent I would either not define the type – or use it here. Since we're not actually adding any receivers to the `Toolset` type I'd be inclined to just use `string` everywhere.